.github/workflows/tests.yml

```diff
@@ -12,97 +12,74 @@ on:
   push:
     branches:
       - master
+      - chore/flakey-flakey-rise-and-shine
   schedule:
-  - cron: "0 0 * * *"
+    - cron: "0 0 * * *"
 
 env:
-  JAVA_DISTRIBUTION: temurin
   MAVEN_OPTS: >-
     -Dmaven.resolver.transport=native
     -Daether.connector.connectTimeout=300000
     -Daether.connector.requestTimeout=300000
 
 jobs:
-  build:
+  test-linux:
     runs-on: ${{ matrix.os }}
+    name: Tests with Java ${{ matrix.java }} on ${{ matrix.os }}
     strategy:
       matrix:
-        java: [11, 17, 20, 21-ea]
-        os: [ubuntu-latest, windows-latest]
-        exclude:
-          - os: windows-latest
-            java: 17
-          - os: windows-latest
-            java: 20
-          - os: windows-latest
-            java: 21-ea
-
-
+        java: [11, 17, 20]
+        os: [ubuntu-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+        with:
+          fetch-depth: 0
+      - name: Setup env
+        uses: ./.github/actions/setup-tests
 
+      - name: Time nix setup
+        run: nix develop .#jdk${{ matrix.java }} --command true
+      - name: Test
+        run: nix develop .#jdk${{ matrix.java }} --command test
 
-    name: Tests with Java ${{ matrix.java }} on ${{ matrix.os }}
+  test-windows:
```

Comment:
Can we also reintroduce a working version of the ea tests? (This could be a separate PR probably)

Reply:
If we build the JDK ourselves (or fetch a built version from somewhere, sure). We can either do that here or as a followup.

Reply:
Okay, let's do that as a followup and potentially test it somewhere first?

Reply:
Alright :)